pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -0,0 +1,845 @@
+		logrus.Infof("start vm import controller")
+		vmimportcontrollerCopy := vmimportcontroller.DeepCopy()
+		*vmimportcontrollerCopy.Spec.Replicas = 1
+		if replicas, err := strconv.Atoi(vmimportcontroller.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
+			*vmimportcontrollerCopy.Spec.Replicas = int32(replicas)
+		}
+		delete(vmimportcontrollerCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+		if _, err := h.deployments.Update(vmimportcontrollerCopy); err != nil {
+			logrus.Warnf("VM Import Controller update error %v", err)
+			return false
+		}
+		return false
+	}
+
+	return true
+}
+
+// check Pod status, if all pods are start, return true
+func (h *Handler) checkPodStatusAndStart() bool {
+	allStarted := true
+
+	if !h.checkPrometheusStatusAndStart() {
+		allStarted = false
+	}
+
+	if !h.checkAltermanagerStatusAndStart() {
+		allStarted = false
+	}
+
+	if !h.checkGrafanaStatusAndStart() {
+		allStarted = false
+	}
+
+	if !h.checkRancherMonitoringStatusAndStart() {
+		allStarted = false
+	}
+
+	if !h.checkVMImportControllerStatusAndStart() {
+		allStarted = false
+	}
+
+	return allStarted
+}
+
+func (h *Handler) checkRancherMonitoringStatusAndStop() bool {
+	// check managedchart fleet-local/rancher-monitoring paused
+	monitoring, err := h.managedChartCache.Get(FleetLocalNamespace, RancherMonitoring)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.Infof("rancher monitoring not found. skip")
+			return true
+		}
+		logrus.Warnf("rancher monitoring get error %v", err)
+		return false
+	}
+
+	logrus.Infof("Rancher Monitoring: %v", monitoring.Spec.Paused)
+	// check pause or not
+	if !monitoring.Spec.Paused {
+		logrus.Infof("stop rancher monitoring")
+		monitoringCopy := monitoring.DeepCopy()
+		monitoringCopy.Annotations[PausedStorageNetworkAnnotation] = "false"
+		monitoringCopy.Spec.Paused = true
+
+		if _, err := h.managedCharts.Update(monitoringCopy); err != nil {
+			logrus.Warnf("rancher monitoring error %v", err)
+			return false
+		}
+		return false
+	}
+
+	return true
+}
+
+func (h *Handler) checkPrometheusStatusAndStop() bool {
+	// check prometheus cattle-monitoring-system/rancher-monitoring-prometheus replica
+	prometheus, err := h.prometheusCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringPrometheus)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.Infof("prometheus not found. skip")
+			return true
+		}
+		logrus.Warnf("prometheus get error %v", err)
+		return false
+	}
+	logrus.Infof("prometheus: %v", *prometheus.Spec.Replicas)
+	// check stopped or not
+	if *prometheus.Spec.Replicas != 0 {
+		logrus.Infof("stop prometheus")
+		prometheusCopy := prometheus.DeepCopy()
+		prometheusCopy.Annotations[ReplicaStorageNetworkAnnotation] = strconv.Itoa(int(*prometheus.Spec.Replicas))
+		*prometheusCopy.Spec.Replicas = 0
+
+		if _, err := h.prometheus.Update(prometheusCopy); err != nil {
+			logrus.Warnf("prometheus update error %v", err)
+			return false
+		}
+		return false
```

Comment:
the `return false` seems to enqueue & drive self for next round check,   but it is unusual to return false when everything is correctly done, please add comment for such purpose

Reply:
added comment